### PR TITLE
Minimum value for Request Throttling config

### PR DESCRIPTION
**Pull request by conorbros**
As per discussion on #565

### CODE REVIEW

File: shotover-proxy/src/transforms/throttling.rs [hunk 1]
```diff
@@ -22,6 +23,12 @@ pub struct RequestThrottlingConfig {
 
 impl RequestThrottlingConfig {
     pub async fn get_source(&self) -> Result<Transforms> {
+        if self.max_requests_per_second < nonzero!(50u32) {
```
Comment (rukai) [inline shotover-proxy/src/transforms/throttling.rs:26]: This should be written as a validate() implementation.
Failures in validation give better information to the user. e.g. Includes the chain name in the error and splits the error across multiple lines to make it more readable.

File: shotover-proxy/src/transforms/throttling.rs [hunk 2]
```diff
@@ -66,4 +69,63 @@ impl Transform for RequestThrottling {
 
         Ok(responses)
     }
+
+    fn validate(&self) -> Vec<String> {
+        if self.max_requests_per_second < nonzero!(50u32) {
+            vec![
+                "RequestThrottling".into(),
```
Comment (rukai) [inline shotover-proxy/src/transforms/throttling.rs:76]: `:` after the transform name

**Review by rukai — APPROVED** [review]
lgtm

**Comment by benbromhead** [comment]
This needs to get documented (not only that there is a minimum, but why... e.g. the rate limiting alg falls apart when you try to do something too low).

**Comment by conorbros** [comment]
Opened a PR for docs #573

**Review by benbromhead — APPROVED** [review]

**Pull request merged**